src/transport/EventLoggingServiceClient.ts

```diff
@@ -54,9 +54,9 @@ class EventLoggingServiceClient {
 
         const wireEventCopy: any = JSON.parse(JSON.stringify(wireEvent));
         wireEventCopy.content.value = `Buffer(${wireEventCopy.content.value.data.length})`
-        Logger.debug(`EventLoggingServiceClient.log sending wireEvent: ${JSON.stringify(wireEventCopy, null, 2)}`);
+        Logger.isDebugEnabled && Logger.debug(`EventLoggingServiceClient.log sending wireEvent: ${JSON.stringify(wireEventCopy, null, 2)}`);
```

Comment:
Can we optimise the above wirecopy?

I.e. as far as I can see the wirecopy is only being used for the debug statements?

If so let's not do a copy (assuming it's even necessary?) unless we are going to log the debug statements.

Reply:
Then I would wrap all of that within an `if (Logger.isDebugEnabled) { ... }`